### DISCUSSION TITLE
Get member results

### DISCUSSION THREAD

**Discussion by scja1991** [Dlubal-Software/RSTAB_Python_Client#28] [Q&A] [answered]
I build an automation to test different designs. At the end I extract results for specific load combinations and specific members by exporting the results to csv-files, read them into my python script and look for the numbers I#m interrested in. The export to csv-file takes a realy long time and get's slower after some iterations. Is there a way to get directly to the internal forces of a member in a specific load case? May be directly with a WS command?

**Comment by heetrojivadiya** [comment]
Hello @scja1991,

Yes, you can get result tables for MemberInternalForces of specific load case or load combination. For that you can assign appropriate enumeration item as loading_type and LC/CO number as loading_no in ResultTables.MembersInternalForces().

for example for Load combination number 2 (CO2):
result = ResultTables.MembersInternalForces(CaseObjectType.E_OBJECT_TYPE_LOAD_COMBINATION, 2, 1)

for example for Load Case number 3 (LC3):
result = ResultTables.MembersInternalForces(CaseObjectType.E_OBJECT_TYPE_LOAD_CASE, 3, 1)

I hope that you are looking for this kind of solutions. Let us know if you are looking for something else.

Best regards,
Heet Rojivadiya

**Reply by scja1991** [reply]
In my case this is not working. I get the error message: 'AttributeError: 'Text' object has no attribute '__keylist__''
I am using RSTAB 9.03.0007 and Python 3.11.3 on an 64-bit Windows machine.

**Reply by heetrojivadiya** [reply, marked as answer]
Hey, 

may I know that which version of RSTAB package you are using? Yesterday, we released the latest version of RSTAB==v1.08.0. In this latest version, this bug is fixed. So, please try the latest version of RSTAB by updating it or if you are using GitHub Desktop then you can clone main branch of this repo for updated source code. Please let us know if this steps fix your issue or not. 

Thanks and best regards,
Heet Rojivadiya

**Reply by scja1991** [reply]
Good morning,

you are right. After an update of the RSTAB api the command is working without an error but I'm still a little bit confused.

result = ResultTables.MembersInternalForces(CaseObjectType.E_OBJECT_TYPE_LOAD_COMBINATION, 2, 1)

Args:
   loading_type (emun): Loading type (LC2 = E_OBJECT_TYPE_LOAD_CASE)
   loading_no (int): Loading Number (CO2 = 2)
   object_no (int): Object number
   model (class, optional): Model instance

I get a rather big list of dictionaries from the command. Afterwards I have to do a lot of work to find the member I am looking for. When I look for the results of a design situation the amount of data is even bigger. What is the Object_no argument for? I thougt this would specify the member of which I want the results... Instead I get all the results. And in which order are the results?

**Reply by dogukankaratas** [reply]
Hi @scja1991,

This function is returning the whole result table, no matter what you have specified. It's a small bug in our system, and we are working on it as high priority.

I'll let you know when it's fixed.

**Reply by scja1991** [reply]
How far is the bugfix @dogukankaratas ?

**Reply by heetrojivadiya** [reply]
Bug is fixed in latest version.
